### PR TITLE
Added minimising chat and questions box

### PR DESCRIPTION
**Pull request by winson8222**
- Add a button to minimise and expand chat box
- Added a question box on "fight"
- Question Box pop up to replace current correct/wrong answer
- Fixed a bug where stats do not show up in the beginning by undoing the abstraction for teamstats listener.

### CODE REVIEW

File: client/src/components/QuestionPopup.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+          left: 5,
+          right: 5,
+          top: 5,
+          bottom: 5,
+        },
+      })
+      .setInteractive();
+
+    // Close button functionality
+    closeButton.on("pointerdown", () => {
```
Comment (shashahchk) [inline client/src/components/QuestionPopup.tsx:71]: very small nit but can we do pointerDown instead

File: client/src/scenes/Battle.ts [hunk 2]
```diff
@@ -110,12 +111,15 @@ export default class Battle extends Phaser.Scene {
       this.setUpDialogBoxListener();
       this.setUpBattleRoundListeners();
 
-      SetUpTeamListeners(this, this.teamUIText);
-      SetUpQuestions(this);
+      // SetUpQuestions(this);
 
       this.events.emit("usernameSet", this.currentUsername);
```
Comment (shashahchk) [inline client/src/scenes/Battle.ts:116]: can change to userNameSet if that is more consistent?

File: client/src/scenes/Battle.ts [hunk 3]
```diff
@@ -133,6 +137,62 @@ export default class Battle extends Phaser.Scene {
     }
   }
 
+  // set up the team listener to display the team  when teams.onChange
+  private setUpTeamListeners() {
+    // on message for "teamUpdate"
+    this.room.onMessage("team-update", (message) => {
```
Comment (shashahchk) [inline client/src/scenes/Battle.ts:143]: perhaps use teamUpdate to standardise

File: client/src/scenes/Battle.ts [hunk 4]
```diff
@@ -409,33 +453,18 @@ export default class Battle extends Phaser.Scene {
...
-        // onclick call back
-      }
-
-      if (button.name === "option4") {
-        console.log("Option 4 clicked");
-        // onclick call back
-      }
-    });
+    this.dialog.on(
+      "button.click",
```
Comment (shashahchk) [inline client/src/scenes/Battle.ts:457]: not sure if this is a property, but if its an event message please update to buttonClick

**Review by pygmychameleon — COMMENTED** [review]
Thanks for adding this functionality. 
Everything works well, just very small nits about naming. 

Can consider not allowing player to move when popup is enabled since it won't be intuitive if one player is fighting the monster but moved to a differnet corner. (other players won't be able to tell by physical location which monster they are fighting)

File: client/src/components/QuestionPopup.tsx (hunk 1 above)
Comment (winson8222) [reply client/src/components/QuestionPopup.tsx:71]: i think this is fixed

File: client/src/scenes/Battle.ts (hunk 3 above)
Comment (winson8222) [reply client/src/scenes/Battle.ts:143]: changed all to team-update now

File: client/src/scenes/Battle.ts (hunk 4 above)
Comment (winson8222) [reply client/src/scenes/Battle.ts:457]: it this also fixed unfortunately

**Pull request merged**